packages/kotti-ui/source/props.test.ts

```diff
@@ -0,0 +1,451 @@
+import { PropOptions } from '@vue/composition-api'
+import { z } from 'zod'
+
+import { makeProps } from './props'
+import { silenceConsole } from './test-utils/silence-console'
+
+const expectDefault = (
+	prop: PropOptions<unknown, boolean>,
+	defaultValue: unknown,
+) => {
+	// @ts-expect-error
+	expect(prop.default()).toEqual(defaultValue)
+	expect(prop.required).toBe(undefined)
+}
+
+const expectRequired = (prop: PropOptions<unknown, boolean>) => {
+	expect(prop.default).toEqual(undefined)
+	expect(prop.required).toBe(true)
+}
+
+const expectValidator = (
+	prop: PropOptions<unknown, boolean>,
+	successCases: unknown[],
+	failCases: unknown[],
+) => {
+	const { validator } = prop
+
+	if (validator === undefined) {
+		expect(validator).not.toBe(undefined)
+		throw new Error('unexpected undefined validator')
+	}
+
+	successCases.forEach((c) => expect(validator(c)).toBe(true))
+	failCases.forEach((c) => expect(validator(c)).toBe(false))
+}
+
+describe('array', () => {
+	beforeAll(silenceConsole)
+
+	it('z.array(z.number())', () => {
```

Comment:
I know it is concise and potentially clear, but I think the test descriptions should be in pure English.
e.g.
`it('tests array of numbers')`

`it('tests nullable array of numbers')` OR `array of numbers or null` 

etc

Reply:
up to you

Reply:
Somewhat addressed. I hope it’s a good compromise